### PR TITLE
Add ipv6 support

### PR DESCRIPTION
**Pull request by ksthiele**
Hello,

this small PR makes the generated example more URI agnostic and now accepts IPv6 or future protocols that are supported by the net package.

### CODE REVIEW

File: codegen/example/example_server.go [hunk 1]
```diff
@@ -274,11 +275,15 @@ func main() {
...
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "invalid URL %#v: %s\n", u.Host, err)
+					os.Exit(1)
+				}
+				u.Host = net.JoinHostPort(h, *{{ $u.Transport.Type }}PortF)
 			} else if u.Port() == "" {
-				u.Host += ":{{ $u.Port }}"
-			}
+				u.Host = net.JoinHostPort(u.Host, ":{{ $u.Port }}")
+			}
```
Comment (raphael) [inline codegen/example/example_server.go]: Nitpick: please remove the extra whitespaces

**Review by raphael — COMMENTED** [review]
This looks great, thank you! left a small nitpick that would be nice to address.

File: codegen/example/example_server.go (hunk 1 above)
Comment (ksthiele) [reply codegen/example/example_server.go]: done

**Comment by raphael** [comment]
Thank you! one last request: could you also back-port this change to the `v2` branch as well please?

**Pull request merged**